src/main/java/org/truffleruby/language/loader/RequireNode.java

```diff
@@ -77,37 +77,60 @@ private boolean requireWithMetrics(String feature, String expandedPathRaw, Dynam
         }
     }
 
+    /** During the require operation we need to load constants marked as autoloaded for the expandedPath (see
+     * {@link FeatureLoader#registeredAutoloads}) and mark them as started loading (via locks). After require we
+     * re-select autoload constants (because their list can be supplemented with constants that are loaded themselves
+     * (i.e. Object.autoload(:C, __FILE__))) and remove them from autoload registry. More details here:
+     * https://github.com/oracle/truffleruby/pull/2060#issuecomment-668627142 **/
     private boolean requireConsideringAutoload(String feature, String expandedPath, DynamicObject pathString) {
         final FeatureLoader featureLoader = getContext().getFeatureLoader();
         final List<RubyConstant> autoloadConstants = featureLoader.getAutoloadConstants(expandedPath);
-        if (autoloadConstants != null) {
-            if (getContext().getOptions().LOG_AUTOLOAD) {
-                String info = autoloadConstants
-                        .stream()
-                        .map(c -> c + " with " + c.getAutoloadConstant().getAutoloadPath())
-                        .collect(Collectors.joining(" and "));
-                RubyLanguage.LOGGER
-                        .info(() -> String.format(
-                                "%s: requiring %s which is registered as an autoload for %s",
-                                RubyContext.fileLine(getContext().getCallStack().getTopMostUserSourceSection()),
-                                feature,
-                                info));
-            }
+        try {
+            if (autoloadConstants != null) {
```

Comment:
I think this `if` should be outside (before) the `try` as before.
That way, if some `autoloadConstantStart` fails (which would be a bug), we'll see it and we won't attempt to `autoloadConstantStop` which might cause an exception as starting failed, and as a result hide the original error.

Reply:
This is a good note - agree 👍